### PR TITLE
New ext ip detect

### PR DESCRIPTION
**Pull request by ysibirski**
## Description
New robust approach to finding External IP of the host under evmapp entrypoint.sh script. Adding support for IPv6 if IPv4 does not exist

## Jira Ticket
https://horizenlabs.atlassian.net/browse/DEVOP-1286

## Changes
evmapp entrypoint.sh script external IP detection

## Breaking Changes
entrypoint.sh run

### CODE REVIEW

File: dockerfiles/evmapp/entrypoint.sh [hunk 1]
```diff
@@ -42,21 +62,33 @@ else
...
-  echo "Error: Failed to detect external IPv4 address, using internal address."
-  SCNODE_NET_DECLAREDADDRESS="$(hostname -I | cut -d ' ' -f1)"
-  SCNODE_NET_DECLAREDADDRESS="${SCNODE_NET_DECLAREDADDRESS%% }"
+  # Falling over to internal IP
+  if [ -z "${SCNODE_NET_DECLAREDADDRESS:-}" ]; then
+    echo "Warning: failed to detect external IPv(4|6) address, using internal address for 'declaredAddress' parameter."
+    SCNODE_NET_DECLAREDADDRESS="$(hostname -I | cut -d ' ' -f1)"
+    SCNODE_NET_DECLAREDADDRESS="${SCNODE_NET_DECLAREDADDRESS%% }"
+  fi
+  export SCNODE_NET_DECLAREDADDRESS
```
Comment (cronicc) [inline dockerfiles/evmapp/entrypoint.sh]: L76 + L90
Single export after the if else block should suffice.

File: dockerfiles/evmapp/entrypoint.sh [hunk 2]
```diff
@@ -42,21 +62,33 @@ else
     export HOME=/root
 fi
 
-# detect external IPv4 address
-SCNODE_NET_DECLAREDADDRESS="$(dig -4 +short +time=2 @resolver1.opendns.com A myip.opendns.com | grep -v ";" || true)"
+# Checking if external IP address is provided by the user via ENV var
+if [ -n "${SCNODE_NET_DECLAREDADDRESS:-}" ]; then
+  # Checking IPv(4|6) address validity
+  ipv4_pattern="^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
+  ipv6_pattern="^((([0-9a-fA-F]){1,4})\:){7}([0-9a-fA-F]){1,4}$"
```
Comment (cronicc) [inline dockerfiles/evmapp/entrypoint.sh]: The regex doesn't support compressed ipv6 notation which both dig and curl return, e.g. `dead:beef::`. I'd suggest to utilize the `ipv6calc` utility (in apt) to validate v4 and v6 addresses as that is more robust.

**Review by cronicc — CHANGES_REQUESTED** [review]

**Review by otoumas — APPROVED** [review]
I can confirm that external ip is properly detected in MacOs.
LGTM!

File: dockerfiles/evmapp/entrypoint.sh (hunk 2 above)
Comment (ysibirski) [reply dockerfiles/evmapp/entrypoint.sh]: This check is only a defensive mechanism from USER input IF someone wants to provide IP address manually, but I can use this ipv6calc to make the check bullet proof :+1:

**Review by cronicc — APPROVED** [review]

**Pull request merged**